_tutorials/vector-search/semantic-highlighting-tutorial.md

```diff
@@ -0,0 +1,258 @@
+{: .note}
+
+## Prerequisites
+
+Before you begin, ensure you have the following:
+
+-  A deployed text embedding model suitable for generating vector embeddings from your text data. See [Choosing a model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/integrating-ml-models/#choosing-a-model).
+-  A deployed sentence transformer or question-answering model specifically designed for semantic highlighting. OpenSearch plans to release pretrained models suitable for this task. For this tutorial, placeholder model IDs are used.
+
+## Step 1: Set up the index
+
+First, create an index to store your text data and its corresponding vector embeddings. You'll need a `text` field for the original content and a `knn_vector` field for the embeddings:
+
+```json
+PUT neural-search-index
+{
+  "settings": {
+    "index.knn": true
+  },
+  "mappings": {
+    "properties": {
+      "text": {
+        "type": "text"
+      },
+      "text_embedding": {
+        "type": "knn_vector",
+        "dimension": 384, 
+        "method": {
+          "name": "hnsw",
+          "space_type": "l2",
+          "engine": "faiss",
+          "parameters": {
+            "ef_construction": 128,
+            "m": 24
+          }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The `dimension` field must contain your chosen embedding model's dimension.
+
+## Step 2: Register and deploy the ML models
+
+You need two types of models for semantic highlighting:
+
+1.  **Text embedding model**: To convert the search query and document text into vectors.
+2.  **Sentence highlighting model**: To analyze the text and identify the most relevant sentences.
+
+First, register and deploy a text embedding model. Model group ID is optional (see [Model access control]({{site.url}}{{site.baseurl}}/ml-commons-plugin/model-access-control/)):
+
+```json
+POST /_plugins/_ml/models/_register?deploy=true
+{
+  "name": "huggingface/sentence-transformers/all-MiniLM-L6-v2",
+  "version": "1.0.1", 
+  "model_format": "TORCH_SCRIPT"
+}
+```
+{% include copy-curl.html %}
+
+This API returns a `task_id` for the deployment operation. Use the [Tasks API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/tasks-apis/get-task/) to monitor the deployment status:
+
+```json
+GET /_plugins/_ml/tasks/<your-task-id>
+```
+{% include copy-curl.html %}
+
+Once the `state` changes to `COMPLETED`, the Tasks API returns the model ID for the deployed model. Note the text embedding model ID; you'll use it in the following steps.
+
+Next, register a pretrained semantic sentence highlighting model:
+
+```json
+POST /_plugins/_ml/models/_register?deploy=true
+{
+  "name": "TODO: placeholder-model-id",
+  "version": "1.0.0",
+  "model_format": "TORCH_SCRIPT"
+}
+```
+{% include copy-curl.html %}
+
+Monitor the deployment status using the Tasks API. Note the semantic highlighting model ID; you'll use it in the following steps.
+
+## Step 3 (Optional): Configure an ingest pipeline 
```

Comment:
Here we use "Configure", but the following sentence uses "create". Please make consistent.

Reply:
Configure = create a pipeline + set it as a default for the index.